app/src/main/java/com/github/se/cyrcle/ui/review/AllReviewsScreen.kt

```diff
@@ -53,176 +64,297 @@ fun Timestamp?.toFormattedDate(): String {
+              .padding(8.dp)) {
+        com.github.se.cyrcle.ui.theme.atoms.Text(
+            text = title,
+            style = MaterialTheme.typography.titleMedium,
+            modifier = Modifier.clickable(onClick = onToggle).padding(8.dp).fillMaxWidth(),
+            color = MaterialTheme.colorScheme.primary,
+            testTag = title)
+
+        if (isExpanded) {
+          content()
+        }
+      }
+}
+
+@Composable
+fun FilterHeader(
+    selectedSortingOption: ReviewSortingOption,
+    onSortingOptionSelected: (ReviewSortingOption) -> Unit
+) {
+  var showFilters by remember { mutableStateOf(false) }
+
+  Column(modifier = Modifier.padding(16.dp)) {
+    Row(
+        modifier = Modifier.fillMaxWidth(),
+        horizontalArrangement = Arrangement.End,
+        verticalAlignment = Alignment.CenterVertically) {
+          Text(
+              text = "Sort Reviews", // "Sort Reviews" or similar title
+              modifier = Modifier.weight(1f),
+              style = MaterialTheme.typography.headlineMedium,
+              color = MaterialTheme.colorScheme.onSurface)
+
+          SmallFloatingActionButton(
+              onClick = { showFilters = !showFilters },
+              icon = if (showFilters) Icons.Default.Close else Icons.Default.FilterList,
+              contentDescription = "Filter",
+              testTag = "ShowFiltersButton")
+        }
+
+    if (showFilters) {
+      FilterSection(
+          title = "Sort Reviews", // "Sort Reviews" title
+          isExpanded = true,
+          onToggle = { /* No toggle needed for always-visible sorting options */}) {
+            SortingOptionSelector(
+                selectedSortingOption = selectedSortingOption,
+                onOptionSelected = onSortingOptionSelected)
+          }
+    }
+  }
+}
+
+@Composable
+fun SortingOptionSelector(
+    selectedSortingOption: ReviewSortingOption,
+    onOptionSelected: (ReviewSortingOption) -> Unit
+) {
+  Column(modifier = Modifier.fillMaxWidth().padding(8.dp)) {
+    Row(
+        modifier =
+            Modifier.fillMaxWidth()
+                .padding(vertical = 4.dp)
+                .clickable { onOptionSelected(ReviewSortingOption.Rating) }
+                .background(
+                    color =
+                        if (selectedSortingOption == ReviewSortingOption.Rating)
+                            MaterialTheme.colorScheme.secondaryContainer
+                        else Color.Transparent,
+                    shape = MaterialTheme.shapes.small)
+                .padding(12.dp)) {
+          Text(
+              text = stringResource(R.string.sort_by_rating),
+              style = MaterialTheme.typography.bodyMedium,
+              color =
+                  if (selectedSortingOption == ReviewSortingOption.Rating)
+                      MaterialTheme.colorScheme.primary
+                  else MaterialTheme.colorScheme.onSurface)
```

Comment:
here don't you think we can keep the text White in both cases and use cerulean rather than secondary container? 

Not an urgent matter, in particular with the future theme refactor

Reply:
I agree that this can and should be handled in the theme refactor, as changes I make on this now would just be overwritten